Add sending KWU study room seat info

JinHoooooou opened the pull request:
chatbot send message about KWU study room seat info to user

---

수정하라고 하신거 수정하긴했는데.. 그 공통부분 뽑아오는게 감이 잘 안와서 아직 못했습니다.

JinHoooooou committed:
Add sending KWU study room seat info

chatbot send message about KWU study room seat info to user

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -112,4 +97,38 @@ public void sendNoticeKwInfo(RequestBodyDTO request) throws Exception {
+			Elements seatInfo = tableRow.getElementsByTag("td");
+			String roomNum = seatInfo.get(0).text();
+			if(roomNum.matches("[1-3]")){
+				printMessage.append("제 " + roomNum + " 열람실 : " + seatInfo.get(2).text() + "/" + seatInfo.get(3).text() + "/" + seatInfo.get(4).text()+ "\n");
```
`printMessage` 로 append 하는 부분은 그런 메세지들 만 보아놓은 Enum class로 관리되면 좋을 것 같음 혹은 상수로 만들어놓고(이름을 붙여서) 쓰던가

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -112,4 +97,38 @@ public void sendNoticeKwInfo(RequestBodyDTO request) throws Exception {
+	@Override
+	public void sendStudyRoomSeatInfo(RequestBodyDTO request) throws IOException, URISyntaxException {
+		String url = "http://mobileid.kw.ac.kr/seatweb/domian5.asp";
+		org.jsoup.nodes.Document doc = Jsoup.connect(url).get();
```
```java
String url = "http://mobileid.kw.ac.kr/seatweb/domian5.asp";
org.jsoup.nodes.Document doc = Jsoup.connect(url).get();
```
이 부분이 공통적으로 반복되는 듯 한데, 반복되는 부분을 없애면 좋을것 같음

JinHoooooou replied:
음.. sendStudyRoomSeatInfo, sendNoticeKwInfo 메소드에서 공통적으로 사용되는 부분을 하나로 합치는게 좋을거같다는 말씀이세요??

TaeHyoungKwon replied:
ㅇㅇㅇ

저부분이 공통적으로 들어가는거 같은데 저부분을 메소드로 추출하면 될듯함
https://underground2.tistory.com/54

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -112,4 +97,38 @@ public void sendNoticeKwInfo(RequestBodyDTO request) throws Exception {
+	}
+	
+	
+	private void BotSendMessageToUser(String message, String channel) throws URISyntaxException {
```
method 명은 동사로
https://howtodoinjava.com/java/basics/java-naming-conventions/
![image](https://user-images.githubusercontent.com/18538418/74310309-3e1ea500-4db0-11ea-8479-735f928a4b0a.png)

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -112,4 +97,38 @@ public void sendNoticeKwInfo(RequestBodyDTO request) throws Exception {
+		
+		Elements table = doc.getElementsByTag("tr");
+		StringBuffer printMessage = new StringBuffer();
+		printMessage.append("도서관 열람실 정보(전체/사용/잔여)\n");
```
얘도 마찬가지
Enum 클래스로 관리 혹은 상수로 관리

ex) 파이썬 기준
`STUDY_ROOM_SEAT_INFO_TITLE_MSG = "도서관 열람실 정보(전체/사용/잔여)\n"`

TaeHyoungKwon replied:
예를 들어서, 이것에 관한 커밋했으면 ,

커밋작업 했다고 댓글로 커밋을 남겨주셈

ex) 0850aa9 에서 작업했습니다 😄

JinHoooooou committed:
Change Jsoup library version

JinHoooooou committed:
Add Junit Test

JinHoooooou committed:
Chage return type of methods

Change void to boolean

JinHoooooou committed:
Add enum class for bot message

JinHoooooou committed:
Delete and add import

JinHoooooou committed:
Code refactoring

Change methods name BotSendMessageToUser to SendBotMessageToUser
Use Jsoup in sendBusInfo instead of using RestTemplate
Change field name more readable

JinHoooooou commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/message/ChatBotMessage.java:
```diff
@@ -0,0 +1,24 @@
+		this.message=message;
+	}
+
+}
```
상수로 표현하라고해서 enum클래스를 만들었는데.. 이렇게 하는게 맞는건지 잘 모르겠네여 ㅜ

TaeHyoungKwon replied:
코멘트 확인했으면 이모지 찍어주셈 위에 엄지척 처럼

TaeHyoungKwon commented:
https://tech.ssut.me/write-a-good-git-commit-message/

커밋 메세지를 동사형으로 + 어떤 것을 리팩토링 했는지 써줘야됨

`Code Refactoring ..` ->  `Refactor code about ...`


![image](https://user-images.githubusercontent.com/18538418/74397419-efc4e100-4e57-11ea-84bc-bdd4880c2164.png)

위와 같이 써주는 것도 좋긴 한데, 기본적으로는 한줄로도 알수있도록 적어주고, 추가 설명이 필요할 때만 아래에 설명

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+class KwInfoBotServiceImplTest {
+
+	@Test
+	void test() throws Exception {
```
메소드명에서 어떤 테스트인지 알 수도있도록 자세히 써주셈

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+		KwInfoBotServiceImpl test = new KwInfoBotServiceImpl();
+		RequestBodyDTO request = new RequestBodyDTO();
+		//assertEquals(true, test.sendStudyRoomSeatInfo(request));
+		assertEquals(true, test.sendNoticeKwInfo(request));
```
`assertTrue` 가 있지 않남?? 고걸로 수정~

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+	void test() throws Exception {
+		KwInfoBotServiceImpl test = new KwInfoBotServiceImpl();
+		RequestBodyDTO request = new RequestBodyDTO();
+		//assertEquals(true, test.sendStudyRoomSeatInfo(request));
```
필요 없는 주석은 삭제!

JinHoooooou replied:
서비스별로 Test를 만드는게 나을까요??

TaeHyoungKwon commented:
커밋 단위는 매우 잘 나눈듯! 👍

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+
+	@Test
+	void test() throws Exception {
+		KwInfoBotServiceImpl test = new KwInfoBotServiceImpl();
```
얘도 마찬가지인데, `test` 라는 것만 봐서는 뭔지 모르겠음

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+	@Test
+	void test() throws Exception {
+		KwInfoBotServiceImpl test = new KwInfoBotServiceImpl();
+		RequestBodyDTO request = new RequestBodyDTO();
```
얘도 좀더 자세히 써주면 좋을듯? `request` 만으로는 먼지 모르곘음

JinHoooooou committed:
Split chat bot message in detail

JinHoooooou committed:
Add GetDataUsingJsoup method and change methods name

JinHoooooou committed:
Add test case of each service

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -36,50 +36,90 @@ public void echoMyMessage(RequestBodyDTO request) throws URISyntaxException {
 		parameters.add("text", message);
 		parameters.add("channel", channel);
-		parameters.add("token", "xoxb-892170117313-902396819217-Jmown4sbBelQMsCz2vXrdM97");
+		parameters.add("token", "xoxb-892170117313-902396819217-ZlU9s06beJ3MMsgNFvKsKN1v");
```
이런 토큰 값은 코드에 직접 올라가면 안됨, 내가 탈취해서 쓸 수도 있기 때문임

사실 이 토큰이야 별로 할게 없지만, 예를들어서, AWS 토큰이라고 하면 이걸로 가격 천만원 이천만원 나오게 할 수 있으니 꼭 알아두시길!

참고로 저런 토큰 값은 시스템 환경변수로 부터 불러오도록 하거나, .json 파일을 만들어 놓고 거기서 토큰 값을 가져오도록 하던가(.json 파일은 .gitignore에 추가) 하면 됨

일단 지금은 머지하고 다시 고쳐보자

TaeHyoungKwon commented on slack_kw_info_chatbot/src/test/java/com/slack/chatbot/service/KwInfoBotServiceImplTest.java:
```diff
@@ -0,0 +1,27 @@
+
+import com.slack.chatbot.dto.RequestBodyDTO;
+
+class KwInfoBotServiceImplTest {
+
+	KwInfoBotServiceImpl testObject = new KwInfoBotServiceImpl();
+	RequestBodyDTO testParameter = new RequestBodyDTO();
+	@Test
+	void SendBusInfoTest() throws Exception {
+		assertTrue(testObject.SendBusInfo(testParameter));
+	}
+	
+	@Test
+	void SendKwuNoticeTest() throws Exception{
+		assertTrue(testObject.SendKwuNotice(testParameter));
+	}
+	
+	@Test
+	void SendKwuStudyRoomSeatTest() throws Exception{
+		assertTrue(testObject.SendKwuStudyRoomSeat(testParameter));
+	}
+}
```
메소드 명을 봤을 때 어떤 테스트 인지 구체적으로 알아야 되는데, 대충만 알 수 있는 듯 이 부분도 책이나 강의 레퍼런스 같은 것들 찾아보면서 해보면 좋을듯 토욜날 같이 짚어보자 일단 수정하지 말고 merge ㄱㄱ

TaeHyoungKwon commented on slack_kw_info_chatbot/src/main/java/com/slack/chatbot/service/KwInfoBotServiceImpl.java:
```diff
@@ -36,50 +36,90 @@ public void echoMyMessage(RequestBodyDTO request) throws URISyntaxException {
+	}
+
+	@Override
+	public boolean SendKwuNotice(RequestBodyDTO request) throws IOException, URISyntaxException {
```
자바 메소드의 네이밍 컨벤션은 `camelCase` 아님?? 카멜 쓰다가 파스칼로 바꾼 이유가 있음?? 이것도 일단은 토욜날 얘기하는 걸로 하고, merge gogo

TaeHyoungKwon commented:
Approve 했다~ Merge GOGO

TaeHyoungKwon commented:
토욜에 하려고했던 것들 안 까먹기 위해서 여기에 추가로 남겨둠

고쳐야 할 것들

1. 중복 코드 제거 - 메소드 추출
2. Secret Key 관리
3. 테스트 메소드 네임 작성 법
4. TDD 맛배기
5. 네이밍 컨벤션
6. 자바 정적 검사 도구 - 띄워쓰기??
7. 적절한 PR 크기 및 PR 나누기
